tgui/packages/tgui/interfaces/common/sorting/SortableTable/index.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { BooleanLike } from "common/react";
+import { Fragment } from "inferno";
+import { useLocalState, useSharedState } from "../../../../backend";
+import { Table, Tooltip } from "../../../../components";
+import { Header } from "../Header";
+import { SearchState, SortableTableHeaderConfig, SortableTableRowProps, SortState } from "./type";
+import { onSortClick, sortAndFilterRows } from "./utils";
+
+
+interface SortableTableHeaderRowProps {
+  config: SortableTableHeaderConfig[],
+  sortState: SortState,
+  setSortBy: (index: number) => void
+}
+
+const SortableTableHeaderRow = (props: SortableTableHeaderRowProps) => {
```

Comment:
Separate this out into its own file (`SortableTableHeaderRow.tsx`) as it has enough complexity in it.